wp/lib/bap_wp/src/output.ml

```diff
@@ -144,16 +144,11 @@ let format_model (model : Model.model) (env1 : Env.t) (env2 : Env.t) : string =
   print_fun_decls fmt model;
   Format.flush_str_formatter ()
 
-let get_mem (m : Z3.Model.model) (env : Env.t) : mem_model option =
+let get_mem (m : Z3.Model.model) (env : Env.t) : mem_model =
   let arch = Env.get_arch env in
-  if Env.is_x86 arch then
-    begin
-      let module Target = (val target_of_arch arch) in
-      let mem, _ = Env.get_var env Target.CPU.mem in
-      Some (extract_array (Constr.eval_model_exn m mem))
-    end
-  else
-    None
+  let module Target = (val target_of_arch arch) in
```

Comment:
Just as a note: we need to eventually fix this, so that we just grab the target from the KB.

A task for another ticket...

Reply:
I agree. There are lots of instances where we get the architecture from the environment to get the target.